docs/adr/0000-whats-the-big-idea.md

```diff
@@ -0,0 +1,151 @@
+# Write and maintain a custom tool for generating foreign-language bindings to rust code.
+
+* Status: accepted
+* Deciders: rfkelly, linacambridge, eoger, thomcc
+* Date: 2020-07-01 (or thereabouts)
```

Comment:
This document captures my take on the high-level "why are we doing this, rather than doing that" of the project; here's a [rendered view](https://github.com/mozilla/uniffi-rs/blob/docs-and-adrs/docs/adr/0000-whats-the-big-idea.md).

I'm tentatively hopeful that the things I'm saying here and uncontroversial, but if you don't agree with things in this doc, that's really useful information that I think we should surface and discuss! So, despite my labeling of the doc as "status: accepted", please don't hesitate to review this as though it were a live discussion.